internal/ptr.go

```diff
@@ -22,9 +26,10 @@ func NewStringPointer(str string) Pointer {
 		return Pointer{}
 	}
 
-	// The kernel expects strings to be zero terminated
-	buf := make([]byte, len(str)+1)
-	copy(buf, str)
+	p, err := unix.BytePtrFromString(str)
+	if err != nil {
+		return Pointer{}
```

Comment:
Should we return the error here?

Reply:
Well, I tried that initially, but didn't like it. `NewStringPointer` is used in a few struct literal decls, and they'd have to be moved out for error checking. `unix.BytePtrFromString`'s only error is  `-EINVAL` if the input string contains a `\0`. I'm not sure if that's worth protecting against.